javascript/ql/src/semmle/javascript/frameworks/NodeJSLib.qll

```diff
@@ -1134,4 +1134,48 @@ module NodeJSLib {
       result = moduleImport().getAPropertyRead(member)
     }
   }
+
+  /**
+   * Provides predicates for working with the "node-fetch" module and its platform-specific instances as a single module.
+   */
+  module Fetch {
+    /**
+     *  Gets a node that imports the "node-fetch" module, or one of its platform-specific instances.
+     */
+    DataFlow::SourceNode moduleImport() {
+      result = DataFlow::moduleImport(["node-fetch", "cross-fetch", "isomorphic-fetch"])
+      or
+      result = DataFlow::globalVarRef("fetch") // https://fetch.spec.whatwg.org/#fetch-api
+    }
+
+    /**
+     * Gets an instance of the `Headers` class.
+     */
+    private DataFlow::NewNode header() {
+      result = moduleImport().getAConstructorInvocation("Headers")
+      or
+      result = DataFlow::globalVarRef("Headers").getAnInstantiation() // https://fetch.spec.whatwg.org/#headers-class
+    }
+
+    /** An expression that is passed as `http.request({ auth: <expr> }, ...)`. */
```

Comment:
Can this docstring be more abstractly descriptive? Looking at the implementation, I do not even see the "auth" of `{ auth: <Expr> }`, which is confusing. 
```suggestion
    /** An expression that is used as a credential in fetch-request. */
```

Reply:
I copy pasted from another `CredentialsExpr`, and I forgot to change the docstring.